catalog/transformer/catalog-transformer-service-atom/src/main/java/ddf/catalog/transformer/response/query/atom/AtomTransformer.java

```diff
@@ -233,156 +223,177 @@ public BinaryContent transform(SourceResponse sourceResponse, Map<String, Serial
       startIndex.setText(Integer.toString(sourceResponse.getRequest().getQuery().getStartIndex()));
     }
 
-    for (Result result : sourceResponse.getResults()) {
-
-      Metacard metacard = result.getMetacard();
-
-      if (metacard == null) {
-        continue;
-      }
-
-      Entry entry = feed.addEntry();
-
-      String sourceName = DEFAULT_SOURCE_ID;
-
-      if (result.getMetacard().getSourceId() != null) {
-        sourceName = result.getMetacard().getSourceId();
-      }
+    if (getCount(sourceResponse) != 0) {
+      sourceResponse.getResults().stream().forEach(r -> addSingleResult(currentDate, feed, r));
+    }
 
-      Element source =
-          entry.addExtension(new QName(FEDERATION_EXTENSION_NAMESPACE, "resultSource", "fs"));
+    byte[] bytes = createOutputStream(feed);
+    return new BinaryContentImpl(new ByteArrayInputStream(bytes), MIME_TYPE);
+  }
 
-      /*
-       * According to the os-federation.xsd, the resultSource element text has a max length of
```

Comment:
❓ Why was this behavior removed?

Reply:
Not removed, just moved below.  many complaints about method complexity resulting in some refactoring

Reply:
I did see quite a bit of refactoring, guess I missed some. 😃